filebeat/harvester/log.go

```diff
@@ -44,6 +44,29 @@ func (h *Harvester) Harvest() {
 		return
 	}
 
+	go func() {
+		select {
+		case <-h.prospectorDone:
+			close(h.done)
+		case <-h.done:
+			return
+		}
+
+	}()
```

Comment:
The reason here is that this go loop can close the harvester based on a channel. The prospectorDone channel can't be used for that as the same is used for all harvester, means all harvesters are stopped, not just this one.